File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/VolleyUtil.java
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.utils;

import android.content.Context;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

public class VolleyUtil
{
  private static RequestQueue queue;

  public static void init(Context context)
  {
    if (queue == null)
      queue = Volley.newRequestQueue(context);
  }

  public static RequestQueue getQueue()
  {
    return queue;
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/VolleyUtil.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.utils

import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley
import org.dolphinemu.dolphinemu.DolphinApplication

object VolleyUtil {
  @JvmStatic
  val queue: RequestQueue by lazy(LazyThreadSafetyMode.SYNCHRONIZED) {
    Volley.newRequestQueue(DolphinApplication.getAppContext())
  }
}
